gusto/time_discretisation/time_discretisation.py

```diff
@@ -479,6 +480,40 @@ def setup(self, equation, apply_bcs=True, *active_labels):
             logger.warning(message)
             self.solver_parameters['snes_type'] = 'newtonls'
 
+            # TODO: move this outside the if-else and just set the fieldsplit_type here
+            if len(self.x0.subfunctions) > 1:
```

Comment:
Posting here for posterity: you suggested (offline) that we could store the mass-matrix solve parameters in a dictionary so that they could be imported in multiple places

Reply:
Added to `solvers/parameters.py`